cmd/roy/harvest_wikidata.go

```diff
@@ -73,3 +101,83 @@ func harvestWikidata() error {
 	)
 	return nil
 }
+
+// checkWikibaseURL guides the user to set the Wikibase/Wikimedia server
+// URL when using a custom endpoint. This URL should be a valid
+// Wikimedia URL. Roy will connect to the Wikimedia API via this URL.
+func checkWikibaseURL() error {
+	if *harvestWikidataWikibaseURL == config.WikidataWikibaseURL() {
+		return fmt.Errorf(
+			"Wikibase server URL for '%s' needs to be configured, can't be: '%s'",
+			*harvestWikidataEndpoint,
+			*harvestWikidataWikibaseURL,
+		)
+	}
+	return nil
+}
+
+// TODO: move these to configuration package. Return errors to the caller.
```

Comment:
Okay, so we'll have a commit to look at with all this functionality in the config. We remove the logging because it's a package and treat errors as errors. Initially, this all looks much cleaner and less complex. This seems to be in-keeping with a fail-early paradigm.

Reply:
https://github.com/ross-spencer/siegfried/pull/2/commits/292fa8a81a1f14ac3dcc95bcb694fa5e0124ded2